server/datastore/mysql/apple_mdm.go

```diff
@@ -448,6 +471,38 @@ func upsertMDMAppleHostDisplayNamesDB(ctx context.Context, tx sqlx.ExtContext, h
 	return nil
 }
 
+func upsertMDMAppleHostMDMInfoDB(ctx context.Context, tx sqlx.ExtContext, serverURL string, fromSync bool, hostIDs ...uint) error {
+	result, err := tx.ExecContext(ctx, `
+		INSERT INTO mobile_device_management_solutions (name, server_url) VALUES (?, ?)
+		ON DUPLICATE KEY UPDATE server_url = VALUES(server_url)`,
+		serverURL, "fleet")
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "last insert id mdm apple host")
+	}
+
+	mdmID, err := result.LastInsertId()
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "last insert id mdm apple host")
+	}
+
+	// if the device is coming from the DEP sync, we don't consider it enrolled
+	// yet.
+	enrolled := !fromSync
```

Comment:
How does this relate to the case where a device is manually enrolled into MDM but it the orbit/osquery agent has not yet enrolled to Fleet? I believe we wanted to also treat that host as "pending", correct?

Reply:
that doesn't match my understanding, `enrolled` in this context is exclusive to MDM and by this point the host is effectively enrolled.

Reply:
Right, but we've also been talking about how this table relates to filtering by `mdm_enrollment_status` for the list hosts API endpoint. Currently, the API says parameter can be one of three values, all of which are derived from combinations of different booleans in the `host_mdm` table.

mdm_enrollment_status | string | query | The mobile device management (MDM) enrollment status to filter hosts by. Can be one of 'manual', 'automatic' or 'unenrolled'.
-- | -- | -- | --


And now that we're talking about adding `mdm_enrollment_status=pending` to the mix. In that context, we've said that pending means assigned to the MDM server in ABM or enrolled with MDM server but not enrolled in Fleet.

Reply:
> And now that we're talking about adding mdm_enrollment_status=pending to the mix. In that context, we've said that pending means assigned to the MDM server in ABM or enrolled with MDM server but not enrolled in Fleet.

I think that's the origin of the confusion, as I understand it, `mdm_enrollment_status=pending` means assigned to the MDM server in ABM, but not enrolled yet, from https://github.com/fleetdm/fleet/issues/8879 (emphasis mine):

> As a user, in the Fleet UI, I want to be able to see *macOS hosts that are waiting to be automatically enrolled a macOS host to Fleet so that I can change this host's team before the device user unboxes the host*.

in terms of the boolean values we have, this is `enrolled = 0 AND installed_from_dep = 1`

wdyt?

Reply:
That makes sense and is consistent with my understanding of how it would for DEP hosts. However, when we worked on the host ingestion, we also treated non-DEP (manual) host as pending during the period between the initial AuthenticateRequest and when osquery enrolls. In most cases, that ought to be a very short time, but we still treated it as pending. For example, if something went wrong with the osquery installation. This is scenario #3 that we tested in the [PR](https://github.com/fleetdm/fleet/pull/9065#pullrequestreview-1230233835).

Reply:
I think those are unrelated scenarios, the issue is explicitly asking to return DEP-only hosts that haven't enrolled into MDM when the filter is applied.

Reply:
I suppose we should double-check with product. I'm not sure that the issue considered this possibility.

Reply:
Otherwise, a user would see a host with incomplete information in the "All hosts" table with no clear indication of why that might be.

Reply:
I don't think the possibility is relevant, `enrolled` in this context only refers to MDM enrollment, so in scenario `3` `enrolled = 1` is expected. The goal of the issue is to give the IT admin a list of hosts that have been purchased but haven't enrolled into DEP yet.

I'll make a note to talk with product about this scenario (enrolled into MDM but not enrolled into osquery) but it is a completely separate scenario.

Reply:
(and filtering for that scenario implies `enrolled = 1` and doing a `JOIN` with `hosts` to check the osquery enrollment status)

Reply:
Alright. Although we should consider how this is going to be used for more than just the filter join. For example, how are we populating the "MDM status" column in the "All hosts" table in the Figma mock for the [issue](https://github.com/fleetdm/fleet/issues/8879#issuecomment-1365453997)?

Reply:
Let's discuss in our meeting this morning.

Reply:
sounds like a plan!